src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -1194,27 +1195,44 @@ namespace winrt::TerminalApp::implementation
     // - args: an array of strings to process as a commandline. These args can contain spaces
     // Return Value:
     // - 0: We should handle the args "in the current window".
-    // - -1: We should handle the args in a new window
+    // - WindowingBehaviorUseNew: We should handle the args in a new window
+    // - WindowingBehaviorUseExistingSameDesktop: We should handle the args "in
+    //   the current window ON THIS DESKTOP"
+    // - WindowingBehaviorUseExisting: We should handle the args "in the current
+    //   window ON ANY DESKTOP"
     // - anything else: We should handle the commandline in the window with the given ID.
     int32_t AppLogic::FindTargetWindow(array_view<const winrt::hstring> args)
     {
         ::TerminalApp::AppCommandlineArgs appArgs;
         const auto result = appArgs.ParseArgs(args);
         if (result == 0)
         {
-            return appArgs.GetTargetWindow();
-
-            // TODO:projects/5
-            //
-            // In the future, we'll want to use the windowingBehavior setting to
-            // determine what happens when a window ID wasn't manually provided.
-            //
-            // Maybe that'd be a special return value out of here, to tell the
-            // monarch to do something special:
-            //
-            // -1 -> create a new window
-            // -2 -> find the mru, this desktop
-            // -3 -> MRU, any desktop (is this not just 0?)
+            const auto parsedTarget = appArgs.GetTargetWindow();
+            if (parsedTarget.has_value())
+            {
+                // parsedTarget might be -1, if the user explicitly requested -1
+                // (or any other negative number) on the commandline. So the set
+                // of possible values here is {-1, 0, ℤ+}
+                return *parsedTarget;
+            }
+            else
+            {
+                // If the user did not provide any value on the commandline,
+                // then lookup our windowing behavior to determine what to do
+                // now.
+                const auto windowingBehavior = _settings.GlobalSettings().WindowingBehavior();
+                switch (windowingBehavior)
+                {
+                case WindowingMode::UseNew:
+                    return WindowingBehaviorUseNew;
+                case WindowingMode::UseExistingSameDesktop:
+                    return WindowingBehaviorUseExistingSameDesktop;
+                case WindowingMode::UseExisting:
+                    return WindowingBehaviorUseExisting;
+                default:
+                    return WindowingBehaviorUseNew;
+                }

```

Comment:
Random idea (this may be awful). Since enums are just numbers, could we have `static_cast<int>(WindowingMode::UseNew)` == `WindowingBehaviorUseNew` (and the same for the rest)?

Reply:
Meh, theoretically yes, but I didn't want to rely on that behavior necessarily. I'm already going to have to have the constants somewhere (since I don't want to have to pull in all of TSM to Remoting just for some constants) so I'm gonna have to do this either here or in Remoting